lib/onebox/helpers.rb

```diff
@@ -118,8 +117,7 @@ def self.fetch_response(location, redirect_limit: 5, domain: nil, headers: nil,
     def self.fetch_content_length(location)
       uri = URI(location)
 
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.is_a?(URI::HTTPS)) do |http|
-        http.open_timeout = Onebox.options.connect_timeout
+      Net::HTTP.start(uri.host, uri.port, open_timeout: Onebox.options.connect_timeout, use_ssl: uri.is_a?(URI::HTTPS)) do |http|
```

Comment:
start will attempt to connect right away, setting the open_timeout in the block is too late